File: Documentation/5.1/Samples/nodejs/client-api/changes/whatIsChangesApi.js
import { DocumentStore } from "ravendb";

const store = new DocumentStore();
const session = store.openSession();

{
    let database;
    //region changes_1
    store.changes([database]);
    //endregion
}

async function example() {
    {
        //region changes_2
        const changes = store.changes();

        await changes.ensureConnectedNow();

        const allDocsChanges = changes.forAllDocuments()
            .on("data", change => {
                console.log(change.type + " on document " + change.id);
            })
            .on("error", err => {
                // handle error
            });

        // ...

        try {
            // application code here
        } finally {
            // dispose changes after use
            if (changes != null) {
                changes.dispose();
            }
        }
        //endregion
    }
}

File: Documentation/6.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Changes/WhatIsChangesApi.cs
using System;
using System.Threading.Tasks;
using Raven.Client.Documents;
using Raven.Client.Documents.Changes;


namespace Raven.Documentation.Samples.ClientApi.Changes
{
    public class WhatIsChangesApi
    {
        private interface IFoo
        {
            #region changes_1
            IDatabaseChanges Changes(string database = null);
            #endregion

            #region changes_ISingleNodeDatabaseChanges
            ISingleNodeDatabaseChanges Changes(string database, string nodeTag);
            #endregion
        }

        public async Task AsyncWhatIsChangesApi()
        {
            using (var store = new DocumentStore())
            {
                #region changes_2
                IDatabaseChanges changes = store.Changes();
                await changes.EnsureConnectedNow();
                var subscription = changes
                        .ForAllDocuments()
                        .Subscribe(change => Console.WriteLine("{0} on document {1}", change.Type, change.Id));
                try
                {
                    // application code here
                }
                finally
                {
                    if (subscription != null)
                        subscription.Dispose();
                }
                #endregion
            }
        }
    }
}
